pkg/provider/middleware.go

```diff
@@ -0,0 +1,130 @@
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/containous/maesh/pkg/k8s"
+	"github.com/containous/maesh/pkg/topology"
+	"github.com/containous/traefik/v2/pkg/config/dynamic"
+)
+
+// MiddlewareBuilder is capable of building middlewares for a service.
```

Comment:
```suggestion
// MiddlewareBuilder builds middlewares for a service.
```

Reply:
Actually, doesn't it build of middlewares _of_ a service?

Reply:
Reopening since second comment was left out

Reply:
Yes, it does. Will replace `for a` by `of a`.